### PR TITLE
Graphs and sessions notebook

### PR DESCRIPTION
**Pull request by awav**
^^^ subj.

### CODE REVIEW

File: doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb [hunk 1]
```diff
@@ -4,9 +4,169 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "TensorFlow gives you a default graph, which you fill with tensors and operations - nodes and edges in the graph respectively. You can find details [here](https://www.tensorflow.org/guide/graphs#building_a_tfgraph) on how to change the default graph to another one or exploit multiple graphs.\n",
+    "\n",
+    "TensorFlow graph is a representation of your computation, to execute your code you need TensorFlow [session](https://www.tensorflow.org/guide/graphs#executing_a_graph_in_a_tfsession). For example, you can think of graphs and sessions as binary sources and actual command running it in  a terminal. Normally, TensorFlow doesn't provide the default session, although GPflow creates default session and you can get it via:"
```
Comment (vdutor) [inline doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: you need Tensorflow -> you need *a* TensorFlow session

File: doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb [hunk 2]
```diff
@@ -4,9 +4,169 @@
...
+   "source": [
+    "gpflow.reset_default_session()\n",
+    "assert session is not gpflow.get_default_session()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can manupulate sessions manually, but you have to make them default for GPflow:"
```
Comment (vdutor) [inline doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: manupulate -> manipulate

File: doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb [hunk 3]
```diff
@@ -4,9 +4,169 @@
...
+      "                                 fixed_shape value  \n",
+      "SquaredExponential/lengthscales         True   1.0  \n",
+      "SquaredExponential/variance             True   1.0  \n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    k = gpflow.kernels.RBF(1)\n",
+    "    print(k)"
```
Comment (vdutor) [inline doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: I think the formatting looks nicer if you don't wrap it in a `print()`

File: doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb [hunk 4]
```diff
@@ -4,9 +4,169 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, TensorFlow variables and tensors for created RBF kernel are initialised whith the session which was closed when python context ended. You can reuse the RBF object by re-inialising it."
```
Comment (vdutor) [inline doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: *the* created RBF kernel

Comment (vdutor) [reply doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: whith -> within

Comment (vdutor) [reply doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: when *the* python context ended

File: doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb (hunk 3 above)
Comment (awav) [reply doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb]: it will not print k then

**Pull request merged**